File: golang/service/pos_service.go
package service

import (
	"casereview-go/model/auth"
	redirect_url "casereview-go/model/redirect-url"
	"casereview-go/model/refund"
)

type PosService interface {
	Refund(request refund.RefundRequest) (refund.RefundResponse, error)
	Auth(request auth.AuthRequest) (auth.AuthResponse, error)
	FetchRedirectionUrl(request redirect_url.RedirectUrlRequest) (redirect_url.RedirectUrlResponse, error)
}

File: kotlin/src/main/kotlin/com/trendyol/posintegrationapi/service/PosService.kt
package com.trendyol.posintegrationapi.service

import com.trendyol.posintegrationapi.model.auth.AuthRequest
import com.trendyol.posintegrationapi.model.auth.AuthResponse
import com.trendyol.posintegrationapi.model.redirect_url.RedirectUrlRequest
import com.trendyol.posintegrationapi.model.redirect_url.RedirectUrlResponse
import com.trendyol.posintegrationapi.model.refund.RefundRequest
import com.trendyol.posintegrationapi.model.refund.RefundResponse

interface PosService {
    fun refund(request: RefundRequest): RefundResponse?

    fun auth(request: AuthRequest): AuthResponse?

    fun fetchRedirectionUrl(request: RedirectUrlRequest): RedirectUrlResponse?
}
